src/self.h

```diff
@@ -45,17 +47,35 @@ typedef struct {
 typedef struct {
 	uint32_t type;
 	uint32_t size;
-	uint32_t unk;
+	uint32_t has_next;
 	uint32_t pad;
 } SCE_controlinfo;
 
-typedef struct {
-	SCE_controlinfo common;
-	char unk[0x100];
-} SCE_controlinfo_5;
+// (from fagdec)
 
-typedef struct {
-	SCE_controlinfo common;
+// type 4, 0x50 bytes
+typedef struct { // 0x40 bytes of data
+	SCE_controlinfo head;
+	uint8_t constant[0x14]; // same for every PSVita/PS3 SELF, hardcoded in make_fself.exe: 627CB1808AB938E32C8C091708726A579E2586E4
+	uint8_t elf_digest[0x20]; // on PSVita: SHA-256 of source ELF file, on PS3: SHA-1
+	uint8_t padding[8];
+	uint32_t min_required_fw; // ex: 0x363 for 3.63
```

Comment:
how about set 360 until adding new arguments?
but if 0x0 has meaning to support every vita version, it's ok